src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.Sdk.BeforeCommon.targets

```diff
@@ -91,4 +92,20 @@ Copyright (c) .NET Foundation. All rights reserved.
     <DefineConstants>$(DefineConstants);$(_FrameworkIdentifierForImplicitDefine)$(_FrameworkVersionForImplicitDefine)</DefineConstants>
   </PropertyGroup>
 
+  <!-- Handle XML documentation file settings -->
+  <PropertyGroup Condition="'$(GenerateDocumentationFile)' == 'true' and '$(DocumentationFile)' == ''">
+    <DocumentationFile>$(IntermediateOutputPath)\$(AssemblyName).xml</DocumentationFile>
+  </PropertyGroup>
+  <PropertyGroup Condition="'$(GenerateDocumentationFile)' == 'false' and '$(DocumentationFile)' != ''">
+    <_InvalidDocumentationFileSettingsError>GenerateDocumentationFile is false but DocumentationFile is '$(DocumentationFile)', which is an invalid combination.  Either set GenerateDocumentationFile to true (or leave it unset), or leave DocumentationFile unset.</_InvalidDocumentationFileSettingsError>
+  </PropertyGroup>
+  <Target Name="_CheckForInvalidDocumentationFileSettings" BeforeTargets="_CheckForInvalidConfigurationAndPlatform">
+    <Error Condition="'$(_InvalidDocumentationFileSettingsError)' != ''" Text="$(_InvalidDocumentationFileSettingsError)" />
+  </Target>
```

Comment:
Or how about this: we assign semantics to all cases, instead of having error cases. If I set explicitly GenerateDocumentationFile to non-true, then don't generate a documentation file. :) This allows me to control the path independently of whether I use it. So I could set the path without conditions and then still  turn generate on or off per configuration.

``` XML
<PropertyGroup Condition="'$(GenerateDocumentationFile)' == ''">
  <GenerateDocumentationFile Condition="'$(DocumentationFile)' == ''">false</GenerateDocumentationFile>
  <GenerateDocumentationFile Condition="'$(DocumentationFile)' != ''">true</GenerateDocumentationFile>
</GenerateDocumentationFile>

<PropertyGroup Condition="'$(GenerateDocumentationFile)' == 'true' and '$(DocumentationFile)' == ''">
  <DocumentationFile>$(IntermediateOutputPath)$(AssemblyName).xml</DocumentationFile>
</PropertyGroup>

<PropertyGroup Condition="'$(GenerateDocumentationFile)' != 'true'">
   <DocumentationFile />
</PropertyGroup>